infrastructure/ssz/src/main/java/tech/pegasys/teku/infrastructure/ssz/tree/TreeNode.java

```diff
@@ -35,6 +37,18 @@
    */
   Bytes32 hashTreeRoot();
 
+  /** Provide a MessageDigest and updates it with the node `hash_tree_root` */
+  default MessageDigest hashTreeRootDigest() {
+    final MessageDigest digest = MessageDigestFactory.createSha256();
+    hashTreeRoot().update(digest);
+    return digest;
+  }
```

Comment:
yeah, seems that saving additional `MessageDigest` creation is definitely better than saving `Bytes32` and the code is cleaner

Reply:
in the new version i use cachedHash as an alias to Data when sizes match (32). Otherwise I allocate without caching.
I suspect doesn't care that much on leafs.